test/natalie/array_test.rb

```diff
@@ -495,6 +495,62 @@
     end
   end
 
+  describe '#first(n)' do
+    it 'should return a new array of the last n elements' do
+      [1, 2, 3].first(1).should == [1]
+      [1, 2, 3].first(2).should == [1, 2]
+      [1, 2, 3].first(3).should == [1, 2, 3]
+      ['a', 'b'].first(1).should == ['a']
+    end
+
+   it 'returns an empty array if n == 0' do
+     [1, 2, 3].first(0).should == []
+     [].first(0).should == []
+   end
+
+    it 'returns empty array if the array is empty' do
+      [].first(2).should == []
+    end
+
+    it 'Should raise ArgumentError on negative n' do
+       r = begin
```

Comment:
Is this how we should test for error and check their type?
Could not find anything else that did this.

Reply:
This is fine. I'm not super picky. There is a shortcut for this though:

```ruby
-> { [].first(-1) }.should raise_error(ArgumentError, 'negative array size')
```